bin/objectbox_model_generator/lib/src/generator.dart

```diff
@@ -1,122 +1,128 @@
+  }
+
+  @override
+  Future<String> generateForAnnotatedElement(
+      Element elementBare, ConstantReader annotation, BuildStep buildStep) async {
+    try {
+      if (elementBare is! ClassElement)
+        throw InvalidGenerationSourceError("in target ${elementBare.name}: annotated element isn't a class");
+      var element = elementBare as ClassElement;
+
+      // load existing model from JSON file if possible
+      String inputFileId = buildStep.inputId.toString();
+      ModelInfo allModels = await _loadModelInfo();
+
+      // optionally add header for loading the .g.json file
+      var ret = "";
+      if (entityHeaderDone.indexOf(inputFileId) == -1) {
+        ret += CodeChunks.modelInfoLoader(ALL_MODELS_JSON);
+        entityHeaderDone.add(inputFileId);
+      }
+
+      // process basic entity (note that allModels.createEntity is not used, as the entity will be merged)
+      ModelEntity readEntity = new ModelEntity(IdUid.empty(), null, element.name, [], allModels);
+      var entityUid = annotation.read("uid");
+      if (entityUid != null && !entityUid.isNull) readEntity.id.uid = entityUid.intValue;
+
+      // read all suitable annotated properties
+      bool hasIdProperty = false;
+      for (var f in element.fields) {
+        int fieldType, flags = 0;
+        int propUid;
+
+        if (f.metadata != null && f.metadata.length == 1) {
+          var annotElmt = f.metadata[0].element as ConstructorElement;
+          var annotType = annotElmt.returnType.toString();
+          var annotVal = f.metadata[0].computeConstantValue();
+          var fieldTypeAnnot = null; // for the future, with custom type sizes allowed: annotVal.getField("type");
+          fieldType = fieldTypeAnnot == null ? null : fieldTypeAnnot.toIntValue();
+          propUid = annotVal.getField("uid").toIntValue();
+
+          // find property flags
+          if (annotType == "Id") {
+            if (hasIdProperty)
+              throw InvalidGenerationSourceError(
+                  "in target ${elementBare.name}: has more than one properties annotated with @Id");
+            if (fieldType != null)
+              throw InvalidGenerationSourceError(
+                  "in target ${elementBare.name}: programming error: @Id property may not specify a type");
+            if (f.type.toString() != "int")
+              throw InvalidGenerationSourceError(
+                  "in target ${elementBare.name}: field with @Id property has type '${f.type.toString()}', but it must be 'int'");
+
+            fieldType = OBXPropertyType.Long;
+            flags |= OBXPropertyFlag.ID;
+            hasIdProperty = true;
+          } else if (annotType == "Property") {
+            // nothing special
+          } else {
+            // skip unknown annotations
+            print(
+                "warning: skipping field '${f.name}' in entity '${element.name}', as it has the unknown annotation type '$annotType'");
```

Comment:
Might there be a valid use-case where a property has a non-ObjectBox annotation (e.g. in Java for JSON marshalling or code inspection)?

Reply:
true, we should allow this, I'll create an issue